### PR TITLE
feat(targeting-context): Add parsed codeowners to API

### PR DESCRIPTION
**Pull request by jangjodi**
Add parsed codeowners to codeowners API

WOR-2779

Add parsed codeowners to codeowners endpoint so that the frontend can display the ownership rules. Currently, only the `raw` rules are exposed to the frontend.

I discovered that the existing `schema` has the parsed codeowner rules, and added the owner IDs to it and renamed "identifier" to "name" for the frontend.

### CODE REVIEW

File: src/sentry/api/endpoints/codeowners/index.py [hunk 1]
```diff
@@ -31,8 +54,18 @@ def get(self, request: Request, project: Project) -> Response:
...
 
+        should_return_schema = features.has(
+            "organizations:streamline-targeting-context", project.organization
+        )
+
         codeowners = list(ProjectCodeOwners.objects.filter(project=project).order_by("-date_added"))
 
+        if should_return_schema and codeowners:
+            for codeowner in codeowners:
+                self.add_owner_id_to_schema(codeowner, project)
```
Comment (jangjodi) [inline src/sentry/api/endpoints/codeowners/index.py:65]: Add owner IDs to the schema for get response

File: src/sentry/api/endpoints/project_ownership.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def validate(self, attrs):
             self.context["ownership"].project.organization,
         ):
             schema = create_schema_from_issue_owners(
-                attrs["raw"], self.context["ownership"].project_id, True
+                attrs["raw"], self.context["ownership"].project_id, add_owner_ids=True
```
Comment (jangjodi) [inline src/sentry/api/endpoints/project_ownership.py:72]: This and line 180 are small renaming fixes from the related ticket for [project_ownership](https://getsentry.atlassian.net/browse/WOR-2675)

File: src/sentry/api/serializers/models/projectcodeowners.py [hunk 3]
```diff
@@ -56,4 +71,10 @@ def serialize(self, obj, attrs, user):
             _, errors = validate_codeowners_associations(obj.raw, obj.project)
             data["errors"] = errors
 
+        if "renameIdentifier" in self.expand and hasattr(obj, "schema") and obj.schema:
+            self.rename_schema_identifier_for_parsing(obj.schema)
+
+        if "addSchema" in self.expand:
+            data["schema"] = obj.schema
```
Comment (jangjodi) [inline src/sentry/api/serializers/models/projectcodeowners.py:78]: Add schema to the put and get response

File: tests/sentry/api/endpoints/test_project_ownership.py [hunk 4]
```diff
@@ -132,16 +132,13 @@ def test_update_with_streamline_targeting(self):
         }
 
     def test_get(self):
```
Comment (jangjodi) [inline tests/sentry/api/endpoints/test_project_ownership.py:134]: Refactored test from related [project_ownership](https://getsentry.atlassian.net/browse/WOR-2675) ticket to make it easier to read

**Review by scttcper — APPROVED** [review]

**Pull request merged**